src/Shared/PackageManagers/MavenProjectManager.cs

```diff
@@ -160,33 +229,65 @@ public override async Task<IEnumerable<string>> EnumerateVersionsAsync(PackageUR
+                {
+                    baseUrl = $"{upstream.GetDownloadRepositoryUrl().EnsureTrailingSlash()}{packageNamespace}/group-index.xml";
+                }
+
+                // Check that the package exists
+                string? content = await GetHttpStringCache(httpClient, baseUrl, useCache);
+
                 List<string> versionList = new();
                 if (string.IsNullOrWhiteSpace(content))
                 {
                     return new List<string>();
                 }
-                
+
                 // Parse the html file.
                 HtmlParser parser = new();
                 AngleSharp.Html.Dom.IHtmlDocument document = await parser.ParseDocumentAsync(content);
-                
-                // Break the version content down into its individual lines. Includes the parent directory and xml + hash files.
-                IEnumerable<string> htmlEntries = document.QuerySelector("#contents").QuerySelectorAll("a").Select(a => a.TextContent);
 
-                foreach (string htmlEntry in htmlEntries)
+                if (upstream == MavenSupportedUpstream.MavenCentralRepository)
+                {
+                    // Break the version content down into its individual lines. Includes the parent directory and xml + hash files.
+                    IEnumerable<string> htmlEntries = document.QuerySelector("#contents").QuerySelectorAll("a").Select(a => a.TextContent);
+
+                    foreach (string htmlEntry in htmlEntries)
+                    {
+                        // Get the version.
+                        if (htmlEntry.EndsWith('/') && !htmlEntry.Equals("../"))
+                        {
+                            var versionStr = htmlEntry.TrimEnd(htmlEntry[^1]);
+                            Logger.Debug("Identified {0} version {1}.", packageName, versionStr);
+                            versionList.Add(versionStr);
+                        }
+                    }
+                }
+                else if (upstream == MavenSupportedUpstream.GoogleMavenRepository)
                 {
-                    // Get the version.
-                    if (htmlEntry.EndsWith('/') && !htmlEntry.Equals("../"))
+                    var packageElement = document.QuerySelector($"{packageName}");
+ 
+                    if (packageElement != null)
                     {
-                        var versionStr = htmlEntry.TrimEnd(htmlEntry[^1]);
-                        Logger.Debug("Identified {0} version {1}.", packageName, versionStr);
-                        versionList.Add(versionStr);
+                        string versions = packageElement.GetAttribute("versions");
+                        var versionsArray = versions.Split(',');
+
+                        foreach (string version in versionsArray)
+                        {
+                            versionList.Add(version.Trim());
+                        }
```

Comment:
Could we consolidate these implementations to read from `maven-metadata.xml`? It seems like most of the repositories listed at https://mvnrepository.com/repos publish that file. Google Maven repository and Maven Central also publish it: https://maven.google.com/android/arch/core/common/maven-metadata.xml

Reply:
With Maven Central, I noticed a couple cases where the version information in `maven-metadata.xml` did not match what was available on the frontend. For example, for the package https://repo1.maven.org/maven2/ant/ant/, the `maven-metadata.xml` only lists 12 versions, whereas 13 versions are available on the frontend. Should we still defer to the `maven-metadata.xml` as the source of truth for versioning?

Reply:
That's very interesting. In that particular example, it looks like version `1.7.0` only has a pom.xml file (and no  .jars). Is `1.7.0` actually usable in a meaningful way?

Reply:
Yeah it seems like it's not actually usable. I ended up using the `maven-metadata.xml` for versioning. If that's not available, then trying to use the directory index.